lib/Db/ProvisioningMapper.php

```diff
@@ -42,30 +45,159 @@ public function getActive() : ?array {
 		$qb = $qb->select('*')
 				->from($this->getTableName())
 				->where($qb->expr()->eq('active', $qb->createNamedParameter('true', IQueryBuilder::PARAM_BOOL)));
-		$active = $qb->execute();
-		$result = $active->fetchAll();
-		$active->closeCursor();
+		$cursor = $qb->execute();
+		$result = $cursor->fetchAll();
+		$cursor->closeCursor();
 		return $result;
 	}
 
-	public function getAll() : ?array {
+	public function findByDomain(string $domain) : ?array {
 		$qb = $this->db->getQueryBuilder();
 		$qb = $qb->select('*')
-				->from($this->getTableName());
-		$all = $qb->execute();
-		$result = $all->fetchAll();
-		$all->closeCursor();
+				->from($this->getTableName())
+				->where($qb->expr()->eq('provisioning_domain', $qb->createNamedParameter($domain)));
+		$cursor = $qb->execute();
+		if (($result = $cursor->fetch()) === false) {
+			return null;
+		}
+		$cursor->closeCursor();
 		return $result;
 	}
 
-	public function getWildcardConfig() : ?array {
+	public function setActive(bool $active, Provisioning $provisioning) : void {
+		$qb = $this->db->getQueryBuilder();
+		$qb = $qb->update($this->getTableName())
+				->set('active', $qb->createNamedParameter($active, IQueryBuilder::PARAM_BOOL))
+				->where($qb->expr()->eq('provisioning_domain', $qb->createNamedParameter($provisioning->getProvisioningDomain())));
+		$qb->execute();
+	}
+
+	public function getAll() : array {
 		$qb = $this->db->getQueryBuilder();
 		$qb = $qb->select('*')
 				->from($this->getTableName())
-				->where($qb->expr()->eq('provisioning_domain', $qb->createNamedParameter('*', IQueryBuilder::PARAM_BOOL)));
-		$wildcard = $qb->execute();
-		$result = $wildcard->fetch();
-		$wildcard->closeCursor();
+				->orderBy('active', 'desc');
+		$cursor = $qb->execute();
+		$result = [];
+		while (($row = $cursor->fetch()) !== false) {
+			$result[] = Provisioning::fromRow($row);
+		}
+		$cursor->closeCursor();
 		return $result;
 	}
+
+	public function getWildcardConfig() : ?Provisioning {
+		$qb = $this->db->getQueryBuilder();
+		$qb = $qb->select('*')
+				->from($this->getTableName())
+				->where($qb->expr()->eq('provisioning_domain', $qb->createNamedParameter(self::WILDCARD)));
+		$cursor = $qb->execute();
+		$result = $cursor->fetch();
+		$cursor->closeCursor();
+		if ($result === false) {
+			return null;
+		}
+		return Provisioning::fromRow($result);
```

Comment:
use ``\OCP\AppFramework\Db\QBMapper::findOneQuery`` then you don't have to deal with the cursor and other repetitive code :)

Reply:
:+1:  it's on the to do for tomorrow